dockstore-common/src/main/java/io/dockstore/common/DescriptorLanguage.java

```diff
@@ -31,6 +31,13 @@
 // @Schema(enumAsRef = true)
 public enum DescriptorLanguage {
     // Add new descriptor language here
+    SMK("SMK", "Snakemake", FileType.DOCKSTORE_SMK, FileType.SMK_TEST_PARAMS, false, true,
+            Set.of("smk", ""), true, true) {
```

Comment:
Why is there an empty string in the set of Snakemake primary descriptor extensions?

Reply:
The root descriptor for Snakemake should be called 'Snakefile' (no suffix), the other descriptors should have the '.smk' suffix.
https://snakemake.readthedocs.io/en/stable/snakefiles/deployment.html
So I was trying to accommodate both.

Reply:
The suffixes appear to be used here:
https://github.com/dockstore/dockstore/blob/719c2d076656c25a2c4ad3a30240dc45ca5cd3df/dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java#L1263
I think this line is checking if the primary descriptor, which per your docs, should be `Snakefile`, matches one of the suffixes.  This test will pass for `Snakefile`, because it ends with the empty string, but it'll also pass for any string, because all strings end with the empty string.  If `Snakefile` is always the primary descriptor, and this is indeed the code that ensures that it exists, we might need to do something different.

Reply:
A good general solution might be to use a string matching library that would allow you to create a `Match` object like:

`match = Match.or(Match.extensions("cwl","yaml"), Match.fileName("fooo"))`

It would Match strings:

`match.isMatch('bar')` => false
`match.isMatch('/abc/fooo')` => true
`match.isMatch('whatever.cwl'` => true

and spit out the encapsulated matching condition as an English string:

`match.getDescription()` => `an extension of 'cwl' or 'yaml' or the file name 'fooo'`

Something like this probably exists already, somewhere.